packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,192 @@
+
+- The hooks pattern for controlled inputs is not very common in ecosystem. Didn't find this being used anywhere apart from downshift-js (in their case, everything is hook). Most popular libraries follow something along the lines of `onChange` + `isSelected`.
+- Initial assumption was that there will be too much of state management on user's end with `onChange` + `isSelected` type of approach but now that I look at the [overall example](#onchange--isselected-), it doesn't seem too complex. It also covers most usecases I could think of like `reset`, reading currently selected options, etc.
+- Additional learning curve of `useSelect` API. Explaining `isSelected` is a lot more easier.
+- Since `useSelect` will return `selectOption` method, consumers have to hook this into their state management.
+
+```jsx
+const { values } = useFormik();
+const { selectOption } = useSelect();
+
+const doSomethingComplex = () => {
+  // ... bunch of code that updates `values`
+  selectOption(values.status);
+};
+
+// As opposed to in `onChange` it will just look like this -
+const { values } = useFormik();
+
+<ActionListItem isSelected={values.status === 'pending'} title="Pending" />;
+```
+
+### Referrences
+
+- Primer
+  - [ActionMenu](https://primer.style/react/ActionMenu#with-selection) (Closest to our approach. Uses `selected` and `onSelect`)
+    ```jsx
+    <ActionList.Item
+      key={index}
+      selected={index === selectedIndex}
+      //
+      onSelect={() => setSelectedIndex(index)}
+    />
+    ```
+- Radix [[CodeSandbox Example](https://codesandbox.io/p/sandbox/holy-cache-5kwcgm?selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A28%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A24%7D%5D&file=%2FApp.jsx)]
+- [Polaris](https://polaris.shopify.com/components/selection-and-input/autocomplete) (They have `selected` object with `onSelect` on top component itself)
+- [downshift-js](https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect#actions) (`useSelect` hook)
+
+## Open Questions
+
+### Are `<ActionListItem />` "selectable" components?
+
+**Context:** We support `isDefaultSelected` on ActionList and plan to support `isSelected` now so that means we are treating ActionList as selectable items. When we have `Button` as trigger, we turn ActionList into menu which is not "selectable" but rather a list of "clickable" items.
+
+- **Pros of keeping ActionList selectable**
+
+  - Selection logic is still applicable in `<AutoComplete />`, or any selectable trigger we plan to add in future
+  - Here we are only talking about "selection" logic on consumer-facing props but selection logic on our end will anyway always be there irrespective of this call as we have single component that acts as `<MenuOption />` and `<SelectOption />`, both. Normally I have seen libraries tackle this in separate components.
```

Comment:
also, how do we do selection logic currently for ActionListItem? based on SelectInput's context?

Reply:
So inside the `onClick` of ActionListItem, we have logic to select that item. We basically use our internal `useDropdown` hook and select that particular item when it's clicked.

It's also more than `onClick`, we also have to select that item when keyboard keys are pressed, we have to set `aria-selected` on it, we have to set `aria-current` to show the current active element.

So overall there is lots of selection logic that will anyway be part of `<ActionListItem />` only. From implementation point, `<ActionListItem />` has to be selectable item because of a11y and functional requirements.


By libraries tackling this separately, I was referring to few libraries who have separate components like this - 

- GitHub
  - [SelectPanel](https://primer.style/react/SelectPanel)
  - [ActionMenu](https://primer.style/react/ActionMenu)
- Radix
  - [DropdownMenu](https://www.radix-ui.com/docs/primitives/components/dropdown-menu)
  - [Select](https://www.radix-ui.com/docs/primitives/components/select#select)

In this case, they can handle selection semantics in select components and clickable action items in dropdown / ActionMenu type of components.